From 4e4208d92aaa60b36419353252abdde5aa0137a8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Renaud=20Casenave-P=C3=A9r=C3=A9?= <renaud@casenave-pere.fr>
Date: Wed, 13 May 2026 22:26:56 +0200
Subject: [PATCH] Remove incorrect and useless gmp version check

---
 CMakeLists.txt | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 20dc3533..84ae1b00 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -231,17 +231,6 @@ macro(find_req_library_and_header _header_var _header _lib_var _lib)
 endmacro()
 
 find_req_library_and_header(GMP_PATH gmp.h GMP_LIB gmp)
-if(GMP_PATH AND EXISTS "${GMP_PATH}/gmp.h")
-  file(STRINGS "${GMP_PATH}/gmp.h" gmp_header_str REGEX "^#define[\t ]+__GNU_MP_VERSION(_MINOR|_PATCHLEVEL)?[\t ]+[0-9]+")
-  string(REGEX REPLACE ".*#define __GNU_MP_VERSION[\t ]+([0-9]+).*" "\\1" GMP_VERSION_MAJOR "${gmp_header_str}")
-  string(REGEX REPLACE ".*#define __GNU_MP_VERSION_MINOR[\t ]+([0-9]+).*" "\\1" GMP_VERSION_MINOR "${gmp_header_str}")
-  string(REGEX REPLACE ".*#define __GNU_MP_VERSION_PATCHLEVEL[\t ]+([0-9]+).*" "\\1" GMP_VERSION_PATCH "${gmp_header_str}")
-  unset(gmp_header_str)
-  set(GMP_VERSION_STRING "${GMP_VERSION_MAJOR}.${GMP_VERSION_MINOR}.${GMP_VERSION_PATCH}")
-  if("${GMP_VERSION_STRING}" VERSION_LESS ${Required_Gmp_Version})
-    message(FATAL_ERROR "Ledger requires GMP version ${Expected_Gmp_Version} or greater, but only GMP version ${GMP_VERSION_STRING} was found")
-  endif()
-endif()
 
 find_req_library_and_header(MPFR_PATH mpfr.h MPFR_LIB mpfr)
 if(MPFR_PATH AND EXISTS "${MPFR_PATH}/mpfr.h")
-- 
2.51.0

